composer/utils/object_store/gcs_object_store.py

```
@@ -0,0 +1,210 @@
+        from composer.utils import parse_uri
+        backend, self.bucket_name, self.prefix = parse_uri(gcs_root_dir)
+        if backend == '':
+            raise ValueError(f"gcs_root_dir ({gcs_root_dir}) doesn't have a valid format")
+        self.prefix = self.prefix.lstrip('/')
+
+        try:
+            self.bucket = self.client.get_bucket(self.bucket_name, timeout=60.0)
+        except Exception as e:
+            _reraise_gcs_errors(gcs_root_dir, e)
+
+    def get_key(self, object_name: str) -> str:
+        return f'{self.prefix}{object_name}'
+
+    def get_uri(self, object_name: str) -> str:
+        return f'gs://{self.bucket_name}/{self.get_key(object_name)}'
+
+    def get_object_size(self, object_name: str) -> int:
+        """Retrieves the size of an object stored in the cloud storage bucket.
+
+        Args:
+            object_name (str): The name of the object in the cloud storage bucket whose size is to be retrieved.
+
+        Returns:
+            int: The size of the object in bytes.
+
+        Raises:
+            FileNotFoundError: If the specified object does not exist in the cloud storage bucket.
+            Exception: If an error occurs while trying to retrieve the object's size.
+        """
+        from google.cloud.storage import Blob
+
+        key = self.get_key(object_name)
+        blob_exists = Blob(bucket=self.bucket, name=key).exists(self.client)
+        if not blob_exists:
+            raise FileNotFoundError(f'{object_name} not found in {self.bucket_name}')
+        try:
+            key = self.get_key(object_name)
+            blob = self.bucket.get_blob(key)
+        except Exception as e:
+            _reraise_gcs_errors(self.get_uri(object_name), e)
+
+        return blob.size  # size in bytes
+
+    def upload_object(self,
+                      src: Union[str, pathlib.Path],
+                      dest: str = '',
+                      callback: Optional[Callable[[int, int], None]] = None):
+
+        del callback
```

maybe we should raise an error if callback is not none since it is not supported here